### ISSUE TITLE
'yaml-files' directive not working for directory

### ISSUE THREAD

**Issue by andrewimeson** [adrienverge/yamllint#390]
yamllint 1.26.1 does not match files in a directory like I expect it to. I have created a [PR in my demonstration repo](https://github.com/andrewimeson/ci-troubleshooting/pull/3) to illustrate the problem, but I'll summarize here.

If I have a YAML file in `group_vars/i_am_a_y-a-m-l_file` which has no file extension. All files under `group_vars/` are assumed to be YAML files,  but without extensions. I try and instruct yamllint to treat them as such using a variety of different directives, but none of them result in yamllint checking the file.

file: `.yamllint`

```yaml
---
extends: default
ignore: |
  .venv/
  .git/
yaml-files:
  - '*.yaml'
  - '*.yml'
  # This works
  - '*.yamaml'
  # None of these work for matching non ".yml" ".yaml" files in group_vars/
  - group_vars
  - group_vars/
  - /group_vars/
  - ./group_vars/
  - ./group_vars
  - group_vars/*
  - group_vars/**
  - ./group_vars/*
  - ./group_vars/**
```

file: `group_vars/i_am_a_y-a-m-l_file`

```yaml
---
this_file_has_problems: TRUE
```

file: `group_vars/bad.yamaml`

```yaml
# yamllint will lint this file, because of the extension match
# it won't lint it without that, even though it's in a directory that should
# match.
has_problems: TRUE
```

Running yamllint, it correctly takes on my custom file extension `.yamaml` (🐬🐻❓), but it doesn't lint `i_am_a_y-a-m-l_file`

```console
$ yamllint .
./group_vars/bad.yamaml
  4:1       warning  missing document start "---"  (document-start)
  4:15      warning  truthy value should be one of [false, true]  (truthy)
```

**Comment by andrewimeson** [comment]
I have a workaround that works, `git ls-files group_vars/ | xargs yamllint`

**Comment by adrienverge** [comment]
Hello @andrewimeson, thanks for the report. Could this be a duplicate of https://github.com/adrienverge/yamllint/issues/334? If yes, could you check whether the problem occurs with version 1.23?

**Comment by andrewimeson** [comment]
@adrienverge You are correct, it is a duplicate of #334. I tried with yamllint 1.23.0, and it works the way I expect it to. I'll close this so we only track it in one place.
